File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    answer = 0
    
    N = len(numbers)
    
    for bit in range((1<<N) -1):
        tmp = 0
        for i in range(N):
            if bit & 1 << i:
                tmp += numbers[i]
            else:
                tmp -= numbers[i]
        if tmp == target:
            answer += 1  
    
    
    
    
    return answer

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.cs
using System;

public class Solution {
    int CNT;
    int N;
    int TG;
    int[] NUMS;
    public int solution(int[] numbers, int target) {
        CNT = 0;
        N = numbers.Length;
        TG = target;
        NUMS = numbers;
        DFS(0, 0);
        
        return CNT;
    }
    
    public void DFS(int idx, int cur) {
        if (idx == N) {
            if (cur == TG) {
                CNT++;
            }
            return;
        }
        DFS(idx+1, cur+NUMS[idx]);
        DFS(idx+1, cur-NUMS[idx]);
    }   
    
    
}